lib/galaxy/tools/__init__.py

```diff
@@ -2469,7 +2476,12 @@ def exec_after_process(self, app, inp_data, out_data, param_dict, job=None, **kw
                 # Revert dataset.state to fall back to dataset.dataset.state
                 dataset._state = None
             # Need to reset the peek, which may rely on metadata
-            dataset.set_peek()
+            # TODO: move this into metadata setting, setting the peek requires dataset access,
+            # and large chunks of the dataset may be read here.
+            try:
+                dataset.set_peek()
```

Comment:
How did I miss this, ugh. I need to get those tests running - but I got some good leads when working on remote ITs. Thanks for the fixes!